Sources/X509/RDNAttribute.swift

```diff
@@ -83,9 +83,136 @@ extension RelativeDistinguishedName.Attribute.Value.Storage: Hashable {
     
     @inlinable
     func hash(into hasher: inout Hasher) {
-        // TODO: implement a ASN1UTF8StringView over a String that conforms to RandomAccessCollection and can lazily bytes that are equal to ASN1Any(ASN1UTF8String(string:))
-        // same for ASN1PrintableString. This would allow us to implement `hash(into:)` without allocating, same for default case in `==` above.
-        ASN1Any(self).hash(into: &hasher)
+        switch self {
+        case .printable:
+            // TODO: implement a ASN1PrintableStringView over a String that conforms to RandomAccessCollection and can lazily get bytes that are equal to ASN1Any(ASN1UTF8String(string:))
+            hasher.combine(ASN1Any(self))
+        case .utf8(let string):
+            hasher.combine(String.ASN1UTF8StringView(string: string))
+            
+        case .any(let asn1Any):
+            hasher.combine(asn1Any)
+        }
+    }
+}
+
+extension String {
+    @usableFromInline
+    struct ASN1UTF8StringView {
+        @usableFromInline
+        var string: String
+        
+        @inlinable
+        init(string: String) {
+            self.string = string
+        }
+    }
+}
+
+extension String.ASN1UTF8StringView: RandomAccessCollection {
+    @inlinable
+    var startIndex: Int {
+        0
+    }
+    
+    @inlinable
+    var endIndex: Int {
+        count
+    }
+    
+    @inlinable
+    var count: Int {
+        let utf8Count = self.string.utf8.count
+        // tag + utf8 bytes length + utf8 bytes
+        return 1 + ASN1Length(length: utf8Count).count + utf8Count
+    }
```

Comment:
Can we compute this ahead of time? No need to recompute it each time we calculate `endIndex`.